File: app/src/main/java/com/vibhorsrv/cameraids/reflection/ReflectionProvider.java
package com.vibhorsrv.cameraids.reflection;

import com.vibhorsrv.cameraids.api.ReflectionApi;

import java.lang.reflect.Field;
import java.lang.reflect.Method;

public class ReflectionProvider implements ReflectionApi {
    @Override
    public Field[] getFields(Class<?> aClass) {
        return aClass.getDeclaredFields();
    }

    @Override
    public Method[] getMethods(Class<?> aClass) {
        return aClass.getDeclaredMethods();
    }

    public String getResultFieldName(Class<?> aClass, String prefix, Integer value) {
        for (Field f : getFields(aClass))
            if (f.getName().startsWith(prefix)) {
                try {
                    if (f.getInt(f) == value)
                        return f.getName().replace(prefix, "");
                } catch (IllegalAccessException e) {
                    e.printStackTrace();
                }
            }
        return "";
    }
}

File: app/src/main/java/com/vibhorsrv/cameraids/reflection/ReflectionProvider.kt
package com.vibhorsrv.cameraids.reflection

import com.vibhorsrv.cameraids.api.ReflectionApi
import java.lang.reflect.Field
import java.lang.reflect.Method

class ReflectionProvider : ReflectionApi {
    override fun getFields(aClass: Class<*>): Array<Field> {
        return aClass.declaredFields
    }

    override fun getMethods(aClass: Class<*>): Array<Method?> {
        return aClass.declaredMethods
    }

    fun getResultFieldName(aClass: Class<*>, prefix: String, value: Int): String {
        for (f in getFields(aClass)) if (f.name.startsWith(prefix)) {
            try {
                if (f.getInt(f) == value) return f.name.replace(prefix, "")
            } catch (e: IllegalAccessException) {
                e.printStackTrace()
            }
        }
        return ""
    }
}
